### PR TITLE
Kent/pr 5

### PR DESCRIPTION
**Pull request by kent-mcleod**

### CODE REVIEW

File: src/kernel/thread.c [hunk 1]
```diff
@@ -325,11 +325,12 @@ static void switchSchedContext(void)
 {
     if (unlikely(NODE_STATE(ksCurSC) != NODE_STATE(ksCurThread)->tcbSchedContext) && NODE_STATE(ksCurSC)->scRefillMax) {
         NODE_STATE(ksReprogram) = true;
-        assert(refill_ready(NODE_STATE(ksCurThread->tcbSchedContext)));
-        assert(refill_sufficient(NODE_STATE(ksCurThread->tcbSchedContext), 0));
```
Comment (xurtis) [inline src/kernel/thread.c:322]: Any particular reason not to also have this before?

Comment (kent-mcleod) [reply src/kernel/thread.c:322]: I moved it to be back in the original location.  Moving the guards would be an independent change?

File: src/object/notification.c [hunk 2]
```diff
@@ -101,16 +95,6 @@ void sendSignal(notification_t *ntfnPtr, word_t badge)
                     MCS_DO_IF_SC(tcb, ntfnPtr, {
                         possibleSwitchTo(tcb);
                     })
-#ifdef CONFIG_KERNEL_MCS
-                    if (tcb->tcbSchedContext != NULL && sc_active(tcb->tcbSchedContext)) {
-                        sched_context_t *sc = SC_PTR(notification_ptr_get_ntfnSchedContext(ntfnPtr));
-                        if (tcb->tcbSchedContext == sc && sc_sporadic(sc)) {
```
Comment (kent-mcleod) [inline src/object/notification.c]: I still need to figure out some way handle this case. The refill_unblock_check should happen in possibleSwitchTo unless the vm thread was in the run queue with an SC that is not the current SC so isn't being unblocked at this point.  However this code isn't under verification and  doRemoteVMCheckBoundNotification doesn't even donate an SC at the moment on SMP (but it should).  So I'm leaving it for now until there's consensus whether we want to make the rest of the changes in this commit now or later.

**Comment by kent-mcleod** [comment]
Merged via other PRs.

**Pull request closed without merging**